src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/DestroyStructureTests.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class DestroyStructureTests
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        public struct SomeTestStruct
+        {
+            public int i;
+            public String s;
+        }
+
+        [StructLayout(LayoutKind.Auto)]
+        public struct SomeTestStruct_Auto
+        {
+            public int i;
+        }
+
+        [Fact]
+        public void DestroyStructure()
```

Comment:
Could you split this into multiple [Fact]s--one for negative (ArgumentNullException), the other for positive(generic and non-generic)?

Reply:
Fixed, see in next commit